File: app/models/assessment/InspectionComment.java
// SPDX-FileCopyrightText: 2024 The members of the EXAM Consortium
//
// SPDX-License-Identifier: EUPL-1.2

package models.assessment;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.ManyToOne;
import models.base.OwnedModel;
import models.exam.Exam;

@Entity
public class InspectionComment extends OwnedModel {

    @Column(columnDefinition = "TEXT")
    private String comment;

    @ManyToOne
    private Exam exam;

    public String getComment() {
        return comment;
    }

    public void setComment(String comment) {
        this.comment = comment;
    }

    public Exam getExam() {
        return exam;
    }

    public void setExam(Exam exam) {
        this.exam = exam;
    }
}

File: app/models/assessment/InspectionComment.scala
// SPDX-FileCopyrightText: 2024 The members of the EXAM Consortium
//
// SPDX-License-Identifier: EUPL-1.2

package models.assessment

import jakarta.persistence.{Column, Entity, ManyToOne}
import models.base.OwnedModel
import models.exam.Exam

import scala.compiletime.uninitialized

@Entity
class InspectionComment extends OwnedModel:
  @Column(columnDefinition = "TEXT")
  var comment: String = uninitialized

  @ManyToOne
  var exam: Exam = uninitialized
